src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnectionContext.cs

```diff
@@ -35,6 +35,13 @@ internal class ServiceConnectionContext : ConnectionContext,
             readerScheduler: PipeScheduler.ThreadPool,
             useSynchronizationContext: false);
 
+        private readonly TaskCompletionSource<int> _connectionEndTcs = new TaskCompletionSource<int>(TaskContinuationOptions.RunContinuationsAsynchronously);
```

Comment:
What is this and who triggers it?

Reply:
Sorry, I was working on another PR split from this one these days and failed to review this PR though I updated it several times.

I've updated the code, it will be set after the client connection completed.